### PR TITLE
Nr1786 remove deprecated technologies

### LINKED ISSUE #1786: Remove deprecated technologies
We deprecated support for a few gems in 8.7. No versions of the following gems will be supported as of 9.0.0, so their instrumentation and test files can be deleted: 

- [x] Acts As Solr
- [x] Authlogic
- [x] DataMapper
- [x] Rainbows
- [x] Sunspot

Our public documentation should be updated accordingly.

### PR DESCRIPTION
**Pull request by hannahramadan**
We deprecated support for a few gems in 8.7. No versions of the following gems will be supported as of 9.0.0, so their instrumentation and test files can be deleted:

- [x]  Acts As Solr
- [x]  Authlogic
- [x]  DataMapper
- [x]  Rainbows
- [x]  Sunspot

Closes #1786

### CODE REVIEW

**Review by tannalynn — APPROVED** [review]
🎉

**Review by fallwith — APPROVED** [review]

File: test/multiverse/lib/multiverse/runner.rb [hunk 1]
```diff
@@ -101,7 +101,7 @@ def execute_suites(filter, opts)
       "agent" => %w[agent_only bare config_file_loading deferred_instrumentation high_security no_json json marshalling yajl],
       "background" => %w[delayed_job sidekiq resque],
       "background_2" => ["rake"],
-      "database" => %w[datamapper elasticsearch mongo redis sequel],
```
Comment (kaylareopelle) [inline test/multiverse/lib/multiverse/runner.rb:104]: also surprised this was our only test suite!

**Review by kaylareopelle — APPROVED** [review]
This cleaning feels amazingggggg. thank youuuuuu 

🧼 🛀 🧼 🛀

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,14 @@
   Support for Ruby 2.2 and 2.3 dropped with this release. They are no longer included in our test matrices and are not supported
   for 9.0.0 and onward.
 
+- **Instrumentation dropped for select gems**
```
[the remark this answers is not in the document]
Comment (hannahramadan) [reply CHANGELOG.md:16]: Ah! I like the "Breaking" headline for this. Perhaps in the final review of the 9.0.0 CHANGELOG, we can add some of this? 

I forgot about the docs. I'll update the scope of this ticket and make a docs PR noting updated support (or lack of it)

[the remark this answers is not in the document]
Comment (hannahramadan) [reply CHANGELOG.md:16]: Docs PR created: https://github.com/newrelic/docs-website/pull/11378

**Pull request merged**